stan/math/prim/meta/scalar_seq_view.hpp

```diff
@@ -33,7 +38,7 @@ class scalar_seq_view<
   int size() const { return c_.size(); }
 
  private:
-  C c_;
+  eval_return_type_t<C> c_;
```

Comment:
ooohh okay I think I'm getting it. Looking at this though I think I'd prefer is `eval_return_type_t` just returned the eval'd type and at the definition using `const eval_return_type_t<C>&`. I think that would make `eval_return_type_t` more generic

Reply:
> I think I'd prefer is eval_return_type_t just returned the eval'd type and at the definition using const eval_return_type_t<C>&.

I think I don't understand this. Are you suggesting `eval_return_type_t<T>` to always be a const ref, even if `T` is not a plain type? That would not work for expressions - we would have a dangling ref in such case.

Reply:
No I was mistaken about what eval_return_type_t is doing. What do you think about calling it `eigen_return_type_t<T>`? It feels like it's made for eigen specifically

Reply:
> What do you think about calling it eigen_return_type_t<T>? It feels like it's made for eigen specifically

Or actually just override `return_type_t` for eigen matrices and arrays?

Reply:
> What do you think about calling it eigen_return_type_t? It feels like it's made for eigen specifically

It is. However it is made more specifically to determine return type of the `.eval()` member function so I put that in the name.

> Or actually just override return_type_t for eigen matrices and arrays?

Despite the similar name this has different intentions than `return_type`. `return_type` promotes scalars to doubles, while this determines what type is returned by calling `.eval()`. I think maybe we could find a better name for `return _type`.

Reply:
Let's make an issue to change the name of return_type and we can call this resolved